### ISSUE TITLE
[Docathon] Convert 4 small rST pages to MyST Markdown (autodoc stubs and index)

### ISSUE THREAD

**Issue by svekars** [pytorch/pytorch#182505]
## Description

These four `.rst` files are small — autodoc stubs, a short note, and a toctree index — and can all be handled in a single PR.

| File | What it is |
|------|-----------|
| [`docs/source/cpp_extension.rst`](https://github.com/pytorch/pytorch/blob/main/docs/source/cpp_extension.rst) | Autodoc stub (~13 lines) for `torch.utils.cpp_extension` |
| [`docs/source/nativert.rst`](https://github.com/pytorch/pytorch/blob/main/docs/source/nativert.rst) | Autodoc stub (~12 lines) for `torch.nativert` |
| [`docs/source/cudnn_persistent_rnn.rst`](https://github.com/pytorch/pytorch/blob/main/docs/source/cudnn_persistent_rnn.rst) | Short note (~8 lines) on cuDNN persistent RNNs |
| [`docs/source/community/index.rst`](https://github.com/pytorch/pytorch/blob/main/docs/source/community/index.rst) | Toctree index page (~14 lines) for the Community section |

## Steps

1. **Assign the issue** to yourself using the `/assigntome` comment.

2. **Rename each file using `git mv`:**
   ```bash
   git mv docs/source/cpp_extension.rst docs/source/cpp_extension.md
   git mv docs/source/nativert.rst docs/source/nativert.md
   git mv docs/source/cudnn_persistent_rnn.rst docs/source/cudnn_persistent_rnn.md
   git mv docs/source/community/index.rst docs/source/community/index.md
   ```
   **Important!** Use `git mv` to convert the files — DO NOT delete the `.rst` and create new `.md` files.
   We won't accept PRs with deleted `.rst` files. We want to preserve the git log.

3. **Convert the content** from rST to MyST Markdown syntax. These files are very small so manual conversion is straightforward.

4. **Build the docs locally** as described in the [README](https://github.com/pytorch/pytorch/tree/main/docs).
   Preview locally with `make serve` to ensure the HTML renders correctly.

5. **Run lintrunner** from the root of the repository:
   ```bash
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

6. **Submit a PR** with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:
   - `pull / linux-docs / build-docs-python-false (pull_request)`
   - `linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)`

   Doc tests start after the Docker build, which might take about 30 minutes.
   The doc build itself should take about 25 minutes. When the build is done,
   you can see the preview of your build by clicking
   **📄 Preview Python docs built from this PR** in the pytorch-bot comment.

7. **Label your PR** with: `docathon-2026`, `module: docs`.

8. **Compare the preview** of the pages you converted with the same pages on the main branch.
   The converted pages in preview must look the same as in the main branch.

9. Once the PR is accepted, one of the maintainers will merge it.

## References

- [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown) (wiki)
- [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html)
- [Example PR](https://github.com/pytorch/pytorch/pull/154438)

cc @sekyondaMeta @AlannaBurke

**Comment by svekars** [comment]
@Priyanshu-byte-coder - can you use the assigntome comment above to assign yourself?

**Comment by Priyanshu-byte-coder** [comment]
Hi @svekars! Assigned — thanks for the ping. The PR was failing the `doc-redirects-check` CI because the four renamed files needed entries in `docs/source/redirects.py`. Three of the renames had enough git similarity for the auto-fix to detect them, but `cudnn_persistent_rnn` fell below the rename threshold (content changed too much during RST→MyST conversion) so it needed a manual redirect. I've added all four redirects and pushed the fix to the PR.

**Comment by svekars** [comment]
doc-redirects-check can be ignored
